modules/objdetect/src/aruco/aruco_dictionary.cpp

```diff
@@ -0,0 +1,435 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html
+
+#include <opencv2/imgproc.hpp>
+#include "opencv2/core/hal/hal.hpp"
+
+#include "precomp.hpp"
+#include "aruco_utils.hpp"
+#include "predefined_dictionaries.hpp"
+#include "apriltag/predefined_dictionaries_apriltag.hpp"
+#include <opencv2/objdetect/aruco_dictionary.hpp>
+
+namespace cv {
+namespace aruco {
+
+using namespace std;
+
+Dictionary::Dictionary(): markerSize(0), maxCorrectionBits(0) {}
+
+
+Dictionary::Dictionary(const Mat &_bytesList, int _markerSize, int _maxcorr) {
+    markerSize = _markerSize;
+    maxCorrectionBits = _maxcorr;
+    bytesList = _bytesList;
+}
+
+
+bool Dictionary::readDictionary(const cv::FileNode& fn) {
+    int nMarkers = 0, _markerSize = 0;
+    if (fn.empty() || !readParameter("nmarkers", nMarkers, fn) || !readParameter("markersize", _markerSize, fn))
+        return false;
+    Mat bytes(0, 0, CV_8UC1), marker(_markerSize, _markerSize, CV_8UC1);
+    std::string markerString;
+    for (int i = 0; i < nMarkers; i++) {
+        std::ostringstream ostr;
+        ostr << i;
+        if (!readParameter("marker_" + ostr.str(), markerString, fn))
+            return false;
+        for (int j = 0; j < (int) markerString.size(); j++)
+            marker.at<unsigned char>(j) = (markerString[j] == '0') ? 0 : 1;
+        bytes.push_back(Dictionary::getByteListFromBits(marker));
+    }
+    int _maxCorrectionBits = 0;
+    readParameter("maxCorrectionBits", _maxCorrectionBits, fn);
+    *this = Dictionary(bytes, _markerSize, _maxCorrectionBits);
+    return true;
+}
+
+
+void Dictionary::writeDictionary(Ptr<FileStorage>& fs) {
```

Comment:
> `Ptr<FileStorage>& fs`

We don't need `Ptr` here:
- `FileStorage` is already a smart pointer over implementation.
- no plants to capture lifetime of that object in this API.
- significant part of OpenCV API already uses `FileStorage&`.

Reply:
In this case, the bindings do not work.
```
/home/alexander/all/opencv/exp/opencv/modules/python/src2/cv2_convert.hpp:55:94: error: ‘to’ is not a member of ‘PyOpenCV_Converter<cv::FileStorage, void>’
   55 | bool pyopencv_to(PyObject* obj, T& p, const ArgInfo& info) { return PyOpenCV_Converter<T>::to(obj, p, info); }
```
`FileStorage` correctly bindings only as Ptr.

I can add one more function without Ptr and without CV_WRAP for C++ code only.

[error_log.txt](https://github.com/opencv/opencv/files/10189118/error_log.txt)

Reply:
It is a binding problem.
It should not affect C++ public API.
Such problems should be raised and resolved instead of hiding them.

Or at least binding-specific workaround applied, like [`Algorithm::write()`](https://github.com/opencv/opencv/blob/4.6.0/modules/core/include/opencv2/core.hpp#L3143-L3150).